lib/IDE/CodeCompletion.cpp

```diff
@@ -4232,9 +4232,18 @@ void CodeCompletionCallbacksImpl::addKeywords(CodeCompletionResultSink &Sink) {
   }
 }
 
+#if defined(_MSC_VER)
+class CodeCompletionTypeContextAnalyzer;
+#endif
+
 namespace  {
   class ExprParentFinder : public ASTWalker {
+#if defined(_MSC_VER)
+    // MSVC needed '::' (http://reviews.llvm.org/D4443)
+    friend class ::CodeCompletionTypeContextAnalyzer;
```

Comment:
Does it work portably if you say `friend class ::swift::CodeCompletionTypeContextAnalyzer;` ?